client/src/components/Workflow/workflows.services.ts

```
@@ -105,3 +105,49 @@ export async function getWorkflowInfo(workflowId: string) {
     const { data } = await axios.get(withPrefix(`/api/workflows/${workflowId}`));
     return data;
 }
+
+/**
+ * For dockstore imported IWC workflows, fetch the README file from the GitHub repository.
+ * @param trsId of form `#workflow/github.com/iwc-workflows/<directory>/...`
+ * @param versionId optional version tag (uses `main` branch if not provided)
+ * @returns the README and CHANGELOG file contents
+ */
+export async function fetchDocsForIwcWorkflow(trsId: string, versionId?: string) {
+    try {
+        const parts = trsId.split("/");
+        if (parts.length !== 5) {
+            throw new Error("Invalid trsId");
+        }
+
+        // Ensure that the trsID is for a GitHub workflow at iwc-workflows
+        if (parts[0] !== "#workflow" || parts[1] !== "github.com" || parts[2] !== "iwc-workflows") {
+            throw new Error("Workflow is not from iwc-workflows");
```

It is fine to throw this, right? Instead of just returning from the function.
One way we could get to this point is if a workflow has `source_metadata` but the trs_id is not from iwc for example.
I made sure we rethrow the error (simply does a `console.debug`), and then in `WorkflowAnnotation` I also catch the error, but do nothing.

---

It's ok, but you might be hiding real errors. I would use a regular return with empty values in the expected case of it not being an iwc workflow, and then handle the case that something went wrong with the request and show that to the user ? Or log at an error level at least. Always good to discriminate expected things that won't work from unexpected failures that might require a fix.